server/routes/api/course.js

```diff
@@ -0,0 +1,223 @@
+var verifyUser = require('../../middleware/Token').verifyUser;
+var requireRole = require('../../middleware/Token').requireRole;
+var Course = require('../../models/assignments/Course');
+var Assignment = require('../../models/assignments/Assignment');
+
+module.exports = (app) => {
+    app.post('/api/assignments/createCourse', requireRole("admin"), function(req, res){
+
+    var name = req.body.name;
+    var code = req.body.code;
+    var department = req.body.department;
+    var description = req.body.description;
+    var resourcesUrl = req.body.resourcesUrl;
+    var endDate = req.body.endDate;
+    var startDate = req.body.startDate;
+    var credits = req.body.credits;
+    var hours = req.body.hours;
+    var prof = req.body.professors;
+    var students = req.body.students;
+
+    if(!name){
+        return res.status(400).send({
+            success: false,
+            message: 'Course name required.'
+        });
+    }
+
+    if(!code){
+        return res.status(400).send({
+            success: false,
+            message: 'Course code required.'
+        });
+    }
+
+    if(!department){
+        return res.status(400).send({
+            success: false,
+            message: 'Department required.'
+        });
+    }
+
+    Course.find({
+        code: code
+        // isDeleted: false
+    }, (err, previousCourse) => {
+        if (err){
+            return res.status(500).send({
+                success: false,
+                message: "Error: Server Error"
+            });
+        }
+        if (previousCourse.length > 0){
+            return res.status(409).send({
+                success: false,
+                message: "Error: Course already exists"
+            });
+        }
+    // save the course
+        const newCourse = new Course();
+
+        newCourse.name = name;
+        newCourse.code = code;
+        newCourse.department = department;
+        newCourse.description = description;
+        newCourse.resourcesUrl = resourcesUrl;
+        newCourse.duration.startDate = startDate;
+        newCourse.duration.endDate = endDate;
+        newCourse.details.credits = credits;
+        newCourse.details.hours = hours;
+        newCourse.professors = prof;
+        newCourse.students = students;
+        console.log(newCourse)

```

Comment:
Prof and Students are Schema.Types.ObjectID. There is no way I can send this as part of a request. I shall send a USN(s), comma separated which needs to be searched and added as a student.